### ISSUE TITLE
批量调用线程不释放

### ISSUE THREAD

**Issue by chukai123** [Tencent-Hunyuan/HunyuanOCR#71]
from transformers import AutoProcessor, HunYuanVLForConditionalGeneration
from PIL import Image
import torch
import os
import glob

def clean_repeated_substrings(text):
    """Clean repeated substrings in text"""
    n = len(text)
    if n < 8000:
        return text
    for length in range(2, n // 10 + 1):
        candidate = text[-length:] 
        count = 0
        i = n - length
        
        while i >= 0 and text[i:i + length] == candidate:
            count += 1
            i -= length

        if count >= 10:
            return text[:n - length * (count - 1)]  

    return text

def process_single_image(image_path, processor, model):
    """处理单张图片并返回识别结果"""
    try:
        image_inputs = Image.open(image_path)
    except Exception as e:
        return f"无法打开图片 {image_path}: {str(e)}"

    messages1 = [
        {"role": "system", "content": ""},
        {
            "role": "user",
            "content": [
                {"type": "image", "image": image_path},
                {"type": "text", "text": "检测并识别图片中的文字，将文本坐标格式化输出。"},
            ],
        }
    ]
    messages = [messages1]
    
    texts = [
        processor.apply_chat_template(msg, tokenize=False, add_generation_prompt=True)
        for msg in messages
    ]
    
    inputs = processor(
        text=texts,
        images=image_inputs,
        padding=True,
        return_tensors="pt",
    )
    
    with torch.no_grad():
        device = next(model.parameters()).device
        inputs = inputs.to(device)
        generated_ids = model.generate(** inputs, max_new_tokens=16384, do_sample=False)
    
    # 提取生成的部分
    if "input_ids" in inputs:
        input_ids = inputs.input_ids
    else:
        input_ids = inputs.inputs  #  fallback
    
    generated_ids_trimmed = [
        out_ids[len(in_ids):] for in_ids, out_ids in zip(input_ids, generated_ids)
    ]
    
    output_text = processor.batch_decode(
        generated_ids_trimmed, 
        skip_special_tokens=True, 
        clean_up_tokenization_spaces=False
    )[0]  # 取第一个元素，因为只有一个样本
    
    # 清理重复子串
    cleaned_text = clean_repeated_substrings(output_text)
    
    # 清理显存
    del inputs, generated_ids, generated_ids_trimmed, input_ids
    torch.cuda.empty_cache()
    
    return cleaned_text

def batch_process_images(img_dir, save_dir, model_name_or_path="tencent/HunyuanOCR"):
    """批量处理图片文件夹中的所有图片"""
    # 创建保存目录
    os.makedirs(save_dir, exist_ok=True)
    
    # 加载处理器和模型
    processor = AutoProcessor.from_pretrained(model_name_or_path, use_fast=False)
    model = HunYuanVLForConditionalGeneration.from_pretrained(
        model_name_or_path,
        attn_implementation="eager",
        dtype=torch.bfloat16,
        device_map="auto"
    )
    
    # 获取图片路径列表（支持常见图片格式）
    image_extensions = ['*.jpg', '*.jpeg', '*.png', '*.bmp', '*.gif']
    image_paths = []
    for ext in image_extensions:
        image_paths.extend(glob.glob(os.path.join(img_dir, ext)))
    
    if not image_paths:
        print(f"在目录 {img_dir} 中未找到任何图片文件")
        return
    
    # 批量处理图片
    for img_path in image_paths:
        print(f"正在处理: {img_path}")
        
        # 处理单张图片
        result = process_single_image(img_path, processor, model)
        
        # 保存结果为md文件
        img_name = os.path.splitext(os.path.basename(img_path))[0]
        save_path = os.path.join(save_dir, f"{img_name}.md")
        
        with open(save_path, 'w', encoding='utf-8') as f:
            f.write(f"# 图片 {img_name} 的文字识别结果\n\n")
            f.write(result)
        
        print(f"结果已保存至: {save_path}")
    
    # 最后清理模型占用的显存
    del model, processor
    torch.cuda.empty_cache()
    print("批量处理完成")

# 使用示例
if __name__ == "__main__":
    img_directory = "path/to/your/images"  # 替换为你的图片文件夹路径
    save_directory = "path/to/save/results"  # 替换为结果保存路径
    batch_process_images(img_directory, save_directory)

-----------------------------------------------------------
如上述代码所示，只有第一张正常，之后就报显存不足，但是我的显存充足的，求大佬解答
